sdcm/scan_operation_thread.py

```diff
@@ -490,18 +492,38 @@ def _validate_fullscan_result(
         output = "\n".join([str(i) for i in result])
         if int(result[0].count) <= 0:
             return f"Fullscan failed - count is not bigger than 0: {output}", Severity.ERROR
-        get_query_trace = cmd_result.get_query_trace().events
-        for trace_event in get_query_trace:
-            if dispatch_forward_statement_regex_pattern.search(str(trace_event)):
-                regex_found = True
-                break
         self.log.debug("Fullscan aggregation result: %s", output)
 
-        if not regex_found:
-            self.log.debug("\n".join([str(i) for i in get_query_trace]))
-            self.log.debug("Fullscan aggregation result: %s", output)
-            return "Fullscan failed - 'Dispatching forward_request' message was not found in query trace events", \
-                Severity.WARNING
+        if self.validate_forward_service_requests_counter < 5:
+            self.validate_forward_service_requests_counter += 1
+        else:
+            is_dispatched_forward_request = False
+
+            prometheus = PrometheusDBStats(
+                self.fullscan_params.db_cluster.test_config.tester_obj().monitors.nodes[0].external_address)
+            prometheus_forward_service_requests = prometheus.query(
+                'scylla_forward_service_requests_dispatched_to_other_nodes',
+                self.validate_forward_service_requests_start_time, time.time())
+
+            self.validate_forward_service_requests_counter = 0
+            self.validate_forward_service_requests_start_time = time.time()
+            # expected format is :
+            # [{'metric': {'__name__': 'scylla_forward_service_requests_dispatched_to_other_nodes',
+            # 'instance': '10.4.0.181', 'job': 'scylla', 'shard': '0'}, 'values': [[1690287334.763, '0']]}, ...]
+            for metric in prometheus_forward_service_requests:
+                self.log.info("metric")
+                self.log.info(metric)
+                if metric["values"][0][1] < metric["values"][-1][1]:
```

Comment:
can you make it more verbose what is tested here? 
line of comment or e.g. put these values to variables:
`if forward_service_requests_dispatched_before_query < forward_service_requests_dispatched_after_query`
(if I got it right)

Reply:
logic was removed